js/src/setup-mc/setup-stepper/store-requirements/index.js

```diff
@@ -29,28 +28,25 @@ export default function StoreRequirements() {
 	const adminUrl = useAdminUrl();
 	const { updateGoogleMCContactInformation, saveSettings } = useAppDispatch();
 	const { createNotice } = useDispatchCoreNotices();
-	const { data: initPhoneNumber } = useGoogleMCPhoneNumber();
 	const { data: address } = useStoreAddress();
 	const { settings } = useSettings();
 
+	/**
+	 * Since it still lacking the phone verification state,
+	 * all onboarding accounts are considered unverified phone numbers.
+	 */
+	const [ isPhoneNumberReady, setPhoneNumberReady ] = useState( false );
 	const [ completing, setCompleting ] = useState( false );
-	const [ phoneNumber, setPhoneNumber ] = useState( {
-		isValid: false,
-		isDirty: false,
-	} );
 
 	const handleChangeCallback = ( _, values ) => {
 		saveSettings( values );
 	};
 
 	const handleSubmitCallback = async () => {
 		try {
-			const { isDirty, countryCallingCode, nationalNumber } = phoneNumber;
-			const args = isDirty ? [ countryCallingCode, nationalNumber ] : [];
-
 			setCompleting( true );
 
-			await updateGoogleMCContactInformation( ...args );
+			await updateGoogleMCContactInformation();
```

Comment:
:question: 
It does remove saving phonenumber at all, doesn't it?

Why do we remove it? I think even though, we mocked the verification API, we can here assume it's verified and save I, right?

Or do you plan to save directly on the verification call? 
https://github.com/woocommerce/google-listings-and-ads/issues/962#issuecomment-911157548

The way I've imagined it to work given the Figma UI, is that we store that verification state on the client-side then save the number on "Save".

The downside of that, a user can tweak with the browser state, and submit "Save changes" for an unverified number.
The upside is that UI-wise we will be more prepared for the Google API to come.

However, the same downside remains even if we save on verify in fronted - by sending save request after receiving verification response. I'm not sure if the server-side is prepared to save changes on verify requests.

Reply:
![2021-09-23 11 54 02](https://user-images.githubusercontent.com/17420811/134452439-d7d073a0-8b32-4075-ad48-fdd7042a3ab5.png)

After integrating backend APIs in subsequent PR, it will call [this backend API](https://github.com/woocommerce/google-listings-and-ads/blob/aa86b7f1bb782572c20c81330d0ac35db71d975b/src/API/Site/Controllers/MerchantCenter/PhoneVerificationController.php#L88-L113) when clicking on the "Verify phone number" button. And that backend API calls to [Google's verify API](https://developers.google.com/shopping-content/guides/phoneverification#verifyphonenumber).

After calling Google's verify API with `verificationId`, `verificationCode`, and `phoneVerificationMethod`, it not only returns whether the verification is passed but also updates the phone number to their database if it's passed. And the phone number to be updated was given when requesting a verification code.

With these API and UI designs, there will be no verified phone number waiting to be committed to saving. So the original form processes of phone number update were removed here. [The previously integrated API](https://github.com/woocommerce/google-listings-and-ads/blob/d9bd2c0679d45c4152db21c049351e5320c78134/js/src/data/actions.js#L529-L540) will be removed along with the subsequent PR.

Reply:
Thanks for the detailed explanation and clarification :)